examples/webrx/js/app.js

```diff
@@ -0,0 +1,156 @@
+/*global wx */
+(function () {
+	'use strict';
+    
+    var localStorageKey = 'todos-webrx';
+    var displayModeAll = "all";
+    var displayModeActive = "active";
+    var displayModeCompleted = "completed";
+
+    // represent a single todo item
+    var Todo = function (title, completed) {
+        this.title = wx.property(title);
+        this.completed = wx.property(completed);
+        this.editing = wx.property(false);
+    };
+    
+    // our main view model
+    var ViewModel = function (todos) {
+        // map array of passed in todos to an observableArray of Todo objects
+        this.todos = wx.list(todos.map(function (todo) {
+            return new Todo(todo.title, todo.completed);
+        }));
+
+        this.todos.changeTrackingEnabled = true;
+        
+        this.current = wx.property();  
+        this.showMode = wx.property('all');
+        
+        this.filteredTodos = this.todos.project(function (todo) {
+            switch (this.showMode()) {
+                case displayModeActive:
+                    return !todo.completed();
+                case displayModeCompleted:
+                    return todo.completed();
+                default:
+                    return true;
+            }
+        }.bind(this), this.showMode.changed);
+        
+        // add a new entry, when enter key is pressed
+        this.addCmd = wx.command(function () {
+            var current = this.current().trim();
+
+            if (current) {
+                this.todos.push(new Todo(current));
+                this.current('');
+            }
+        }, this);
+        
+        // remove a single entry
+        this.removeCmd = wx.command(function (todo) {
+            this.todos.remove(todo);
+        }, this);
+                
+        // writeable computed observable to handle marking all complete/incomplete
+        this.completeAllCmd = wx.command(function () {
+            this.todos.forEach(function (todo) {
+                todo.completed(this.remainingCount());
+            }, this);
+        }, this);
+        
+        // remove all completed entries
+        this.removeCompletedCmd = wx.command(function () {
+            this.todos.filter(function (todo) {
+                return todo.completed();
+            }).forEach(function(item) {
```

Comment:
This line but also all the other lines where you've done it `function(a, b)` -> `function (a, b)` please ;)